### ISSUE TITLE
[AutoRoom] [BUG] room creation fails if source channel previously had a bitrate higher than the current limit

### ISSUE THREAD

**Issue by Mathsterk** [PhasecoreX/PCXCogs#116]
Hi, found a bug in autoroom, when a server looses boost and the bitrate limit gets dropped, if the source channel has had a bitrate higher than the one allowed by the lower boost level, the autoroom creation fails when the cog tries to create a room with the "old" bitrate.

In this example, the source had the bitrate set to 256 kbit/s, then the server lost the boost level, so the new max was 128 kbit/s, but the cog tried to create a room with a bitrate of 256 kbit/s, which causes the room creation to fail.

```
redbot    | [2023-05-22 03:31:56] [ERROR] discord.client: Ignoring exception in on_voice_state_update
redbot    | Traceback (most recent call last):
redbot    |   File "/data/venv/lib/python3.11/site-packages/discord/client.py", line 441, in _run_event
redbot    |     await coro(*args, **kwargs)
redbot    |   File "/data/cogs/CogManager/cogs/autoroom/autoroom.py", line 334, in on_voice_state_update
redbot    |     await self._process_autoroom_create(joining.channel, asc, member)
redbot    |   File "/data/cogs/CogManager/cogs/autoroom/autoroom.py", line 448, in _process_autoroom_create
redbot    |     new_voice_channel = await guild.create_voice_channel(
redbot    |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
redbot    |   File "/data/venv/lib/python3.11/site-packages/discord/guild.py", line 1407, in create_voice_channel
redbot    |     data = await self._create_channel(
redbot    |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
redbot    |   File "/data/venv/lib/python3.11/site-packages/discord/http.py", line 744, in request
redbot    |     raise HTTPException(response, data)
redbot    | discord.errors.HTTPException: 400 Bad Request (error code: 50035): Invalid Form Body
redbot    | In bitrate: int value should be less than or equal to 128000.
```

**Comment by PhasecoreX** [comment]
When the boost was lost and the bitrate dropped from 256 to 128, does the source still report 256 in it? It should be copying the source bitrate, but if that's still 256, that's interesting. I think there's a way to get the current server bitrate limit and I'll just have to compare it to the desired source bitrate.

**Comment by PhasecoreX** [comment]
Should be fixed with e61b82d, let me know if it's not (I don't have boosts nor an AutoRoom Source to test it out). Thanks for finding this bug!
